### PR TITLE
Enable Logout on Auth0

### PR DESCRIPTION
**Pull request by steve-keep**
## Summary by CodeRabbit

- **New Features**
	- Introduced `Auth0Logout` component for easier logout management.
	- Added `sidebarFooter` customization capability to the admin UI.

- **Bug Fixes**
	- Updated authentication logic to use a custom `getAuth0Client` function for improved token handling.

- **Refactor**
	- Simplified the logout and login processes within the admin UI components.
	- Removed unnecessary imports and streamlined related functionality.

- **Chores**
	- Adjusted `introspection` settings in server configuration to be environment-dependent.

### CODE REVIEW

File: src/packages/auth-ui-components/src/components/auth-zero/logout/component.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Logout } from '../../logout';
+import { getAuth0Client } from '../client';
+
+export const Auth0Logout = () => {
+	const handleLogout = async () => {
+		const client = await getAuth0Client();
+		await client.logout();
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/auth-ui-components/src/components/auth-zero/logout/component.tsx:7]: This is actually a great idea 😄 

Ill add a toast for any errors!

File: src/packages/auth-ui-components/src/components/logout/component.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { Button, apolloClient, localStorageAuthKey } from '@exogee/graphweaver-admin-ui-components';
+import { LogoutIcon } from '../../assets/16-logout';
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/auth-ui-components/src/components/logout/component.tsx:2]: it is on line 21

File: src/packages/auth-ui-components/src/components/logout/component.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+export const Logout = ({ onLogout }: LogoutProps) => {
+	const navigate = useNavigate();
+
+	const handleOnLogout = async () => {
+		if (onLogout) await onLogout();
+		localStorage.removeItem(localStorageAuthKey);
+		await apolloClient.clearStore();
+		await apolloClient.resetStore();
+		navigate(0);
+	};
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/auth-ui-components/src/components/logout/component.tsx:20]: added with toast!

File: src/packages/auth-ui-components/src/components/auth-zero/logout/component.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/auth-ui-components/src/components/auth-zero/logout/component.tsx:7]: the try catch on this one is one level up

File: src/packages/auth-ui-components/src/components/auth-zero/client.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { Auth0Client, createAuth0Client } from '@auth0/auth0-spa-js';
+import { localStorageAuthKey } from '@exogee/graphweaver-admin-ui-components';
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/auth-ui-components/src/components/auth-zero/client.ts:2]: are you feeling ok @coderabbitai? Its being used on line 22!

File: src/packages/auth-ui-components/src/components/auth-zero/client.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+	try {
+		auth0Client = await createAuth0Client({
+			domain: import.meta.env.VITE_AUTH_ZERO_DOMAIN,
+			clientId: import.meta.env.VITE_AUTH_CLIENT_ID,
+			cache,
+		});
+	} catch (err) {
+		console.error(err);
+		throw new Error('Failed to create Auth0 client');
+	}
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/auth-ui-components/src/components/auth-zero/client.ts:30]: done!

File: src/packages/auth-ui-components/src/components/auth-zero/login/component.tsx [hunk 6]
```diff
@@ -1,19 +1,7 @@
...
-	get: () => undefined,
-	remove: () => {},
-	set: (_: string, value: any) => {
-		const accessToken = value?.body?.access_token;
-		if (accessToken) {
-			localStorage.setItem(localStorageAuthKey, `Bearer ${accessToken}`);
-		}
-	},
-};
+import { getAuth0Client } from '../client';
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/auth-ui-components/src/components/auth-zero/login/component.tsx:4]: its being used twice on line 12 and 19!

**Review by thekevinbrown — APPROVED** [review]

**Pull request merged**